documentation/modules/exploit/unix/webapp/kaltura_unserialize_rce.md

```diff
@@ -0,0 +1,135 @@
+
+Kaltura has a module named keditorservices that takes user input and then use it as an unserialize function parameter. The object constructed is based on the SektionEins Zend code execution POP chain PoC, with a minor modification to ensure Kaltura processes it and the Zend_Log function's __destruct() method is called. Kaltura prior to 11.1.0-2 versions are affected by issue.
+
+**Vulnerable Application Installation Steps**
+
+Kaltura has their own RPM and/or DEB packages to helps us to install it without any issue. Following steps are slightly different than official wiki in order to install vulnerable packages.
+
+Following steps are valid CentOS 6 x64 bit operating system.
+
+1. Install CentOS-6 x64 and run `yum update -y` in order to fetch and install latests packages. Also set hostname something like _kalturahack.dev_ would be wise. Because it will be used during Kaltura installation.
+2. Disable iptables and selinux. 
+```
+iptables -F
+service iptables stop
+chkconfig iptables off
+setenforce permissive
+(selinux will be enabled on reboot unless editing /etc/selinux/config file.)
+```
+3. Install following pre-requisites.
+* PHP 5.3.n, 5.4.n, 5.5.n, 5.6.n [the official distro repo version]
+* MySQL/MariaDB [the official distro repo version]. Please make sure your MySQL is configured not to enforce strict mode.
+* JRE 1.7.n (openjdk can be used)
+
+4. Install Kaltura yum repo.
+```
+rpm -ihv http://installrepo.kaltura.org/releases/kaltura-release.noarch.rpm
+```
+
+5. Kaltura repo configures for latest version by default. We need to change it to one of the vulnerable release. Thus, open `/etc/yum.repos.d/kaltura.repo` file with your favorite text editor and perform following replacement.
+
+Original file. (# lines just ignored)
+```
+[Kaltura]
+name = Kaltura Server
+baseurl = http://installrepo.kaltura.org/releases/latest/RPMS/$basearch/
+gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
+gpgcheck = 1 
+enabled = 1
+
+[Kaltura-noarch]
+name = Kaltura Server arch independent
+baseurl = http://installrepo.kaltura.org/releases/latest/RPMS/noarch
+gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
+gpgcheck = 1
+enabled = 1
+baseurl = http://installrepo.kaltura.org/releases/latest/RPMS/$basearch/
+```
+After changes.
+```
+[Kaltura]
+name = Kaltura Server
+baseurl = http://installrepo.kaltura.org/releases/10.10.0/RPMS/$basearch/
+gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
+gpgcheck = 1 
+enabled = 1
+
+[Kaltura-noarch]
+name = Kaltura Server arch independent
+baseurl = http://installrepo.kaltura.org/releases/10.10.0/RPMS/noarch
+gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
+gpgcheck = 1
+enabled = 1
+baseurl = http://installrepo.kaltura.org/releases/latest/RPMS/$basearch/
+```
+
+6. Install kaltura-server. This will take for a while. 
+```
+yum clean all
+yum install kaltura-server
+```
+
```

Comment:
When Ifollowed this guide, the install failed; I tried again and added the following commands and it worked:
```
/opt/kaltura/bin/kaltura-mysql-settings.sh
service memcached restart
service ntpd restart
chkconfig memcached on
chkconfig ntpd on
```
Offhand, I bet that only the db setup script really mattered.